File: raspberry/SmartHome/arduinodevices/src/main/java/smarthome/raspberry/arduinodevices/controllers/ArduinoOnOff.java
//package smarthome.raspberry.arduinodevices.controllers;
//
//import java.io.IOException;
//
//import smarthome.library.common.ControllerType;
//import smarthome.raspberry.arduinodevices.ArduinoControllerResponse;
//import smarthome.raspberry.arduinodevices.ArduinoDevice;
//
//public class ArduinoOnOff extends ArduinoController implements ArduinoWritable, ArduinoReadable {
//
//    public static final String TAG = ArduinoOnOff.class.getSimpleName();
//
//    public ArduinoOnOff(ArduinoDevice device, String name, ControllerType type,
//                        int indexInArduinoServicesArray) {
//        super(device, name, type, indexInArduinoServicesArray);
//    }
//
//    @Override
//    public ArduinoControllerResponse write(String value) throws IOException {
//        return baseWrite(value);
//    }
//
//    @Override
//    public ArduinoControllerResponse read() throws IOException {
//        return baseRead();
//    }
//}

File: raspberry/arduinodevices/src/main/java/smarthome/raspberry/arduinodevices/domain/controllers/ArduinoOnOff.kt
package smarthome.raspberry.arduinodevices.domain.controllers

import smarthome.library.common.BaseController
import smarthome.library.common.Id
import smarthome.raspberry.arduinodevices.domain.state.BooleanValueState

class ArduinoOnOff(id: Id, name: String, state: BooleanValueState) :
    BaseController(id = id, name = name, state = state), Writeable